libsolidity/parsing/Parser.cpp

```diff
@@ -140,9 +140,16 @@ void Parser::parsePragmaVersion(SourceLocation const& _location, vector<Token> c
 {
 	SemVerMatchExpressionParser parser(_tokens, _literals);
 	auto matchExpression = parser.parse();
+	if (!matchExpression.has_value())
+		m_errorReporter.fatalParserError(
+			1684_error,
+			_location,
+			"Found version pragma, but failed to parse it. "
+			"Please ensure there is a trailing semicolon."
+		);
 	static SemVerVersion const currentVersion{string(VersionString)};
 	// FIXME: only match for major version incompatibility
```

Comment:
Do we actually want this? I am not proposing to do it in this PR, but we could remove this todo if it actually doesn't make sense.

Reply:
Also not sure. Why match for major incomp. only?

Reply:
I think the idea was that it is reported later nicely, but recoverable parser errors make this more complicated, and I think the reporting (here and in SyntaxChecker) are identical, so it is fine.